Paper: The myth of the normal curve and what to do about it

Alcampopiano opened the pull request:
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.


*Editor*: <!--editor-->@cbcunc<!--end-editor-->

*Reviewers*: <!--reviewers-list-->@laserkelvin<!--end-reviewers-list-->

Alcampopiano committed:
adding initial draft

Alcampopiano committed:
fixing bib errors

Alcampopiano committed:
fixing bib errors

Alcampopiano committed:
restored to the way I think is correct despite not building

scoobies commented:
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

Alcampopiano commented:
@scoobies help

scoobies commented:
Hello @Alcampopiano, here are the things you can ask me to do:

```

# List all available commands
@scoobies help

# Show our community Code of Conduct and Guidelines
@scoobies code of conduct

# Add to this issue's reviewers list
@scoobies add @username to reviewers

# Remove from this issue's reviewers list
@scoobies remove @username from reviewers

# Assign a user as the editor of this submission
@scoobies assign @username as editor

# Remove the editor assigned to this submission
@scoobies remove editor

# Add a user to this issue's assignees list
@scoobies add assignee: @username

# Remove a user from this issue's assignees list
@scoobies remove assignee: @username

# Builds paper
@scoobies build paper

# Checks build status
@scoobies build status

# Check the references of the paper for missing DOIs
# Optionally, it can be run on a non-default branch 
@scoobies check references
@scoobies check references from custom-branch-name

# Label issue with: paper
@scoobies mark paper

# Label issue with: needs-review. Remove labels: unready, needs-more-review, pending-comment, ready
@scoobies mark needs review

# Label issue with: needs-more-review. Remove labels: unready, needs-review, pending-comment, ready
@scoobies mark needs more review

# Label issue with: pending-comment. Remove labels: unready, needs-review, needs-more-review, ready
@scoobies mark pending comment

# Label issue with: ready. Remove labels: unready, needs-review, needs-more-review, pending-comment
@scoobies mark ready

# Label issue with: unready. Remove labels: needs-review, needs-more-review, pending-comment, ready
@scoobies mark not ready

# Label issue with: does_not_build:server
@scoobies mark server fail

# Remove labels: does_not_build:server
@scoobies mark server success

```

Alcampopiano commented:
@scoobies mark server fail

Alcampopiano committed:
indent in bib file

Alcampopiano committed:
indent in bib file

Alcampopiano committed:
fixed local build issues

Alcampopiano commented:
@scoobies mark server success

deniederhut commented:
@scoobies mark paper

deniederhut commented:
@scoobies mark needs review

deniederhut commented:
@scoobies build status

scoobies commented:
{"Alcampopiano-2022"=>{"data"=>{"build_output"=>"[*] Updating proceedings build tools in {self.master_repo_path}...\nAlready up to date.\n[*] Check out paper repository...\nCloning into '/tmp/tmplslckxeb'...\nMoving proceedings build tools to temp directory...\n[*] Build the paper...\nBuilding: allan_campopiano\n*** Warning: /tmp/tmplslckxeb/publisher/../output/allan_campopiano/paper_stats.json does not exist.\n", "build_pdf_path"=>"/procbuild/procbuild/../cache/Alcampopiano-2022.pdf", "build_status"=>"success", "build_timestamp"=>"29/05 01:04"}, "status"=>"success"}}

deniederhut commented:
@scoobies assign @cbcunc as editor

scoobies commented:
Assigned! @cbcunc is now the editor

Alcampopiano committed:
fixed typos; added acknowledgements and code output

Alcampopiano committed:
fixed small typo in Acknowledgements

cbcunc commented:
Editor confirms paper builds.

mepa commented:
@scoobies add @ojemebo to reviewers

scoobies commented:
@ojemebo added to the reviewers list!

deniederhut commented:
@scoobies add @laserkelvin to reviewers

scoobies commented:
@laserkelvin added to the reviewers list!

laserkelvin commented:
Thanks @deniederhut! 

@Alcampopiano I'll post comments here by mid-next week before the initial deadline, if I don't get to it sooner.

deniederhut commented:
Hi! We've moved the deadline for first reviews back one week, to June 16th, to give us a little more time to find reviewers for all the papers this year. You can see the updated full timeline at https://github.com/scipy-conference/scipy_proceedings#timeline-for-2022.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+
+    “Everyone believes in the [normal] law of errors, the experimenters because they think it is a mathematical theorem, the mathematicians because they think it is an experimental fact.”
+
+And even when we violate the assumption of the normal curve, we’re still pretty clingy to it
```
> ...we're still pretty clingy to it...

Consider rephrasing to make it less colloquial

Alcampopiano replied:
Thank you for this suggestion. I have reworded accordingly and improved overall clarity in this section via [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/f46499bd0b357b70956619f605c0047f843df7e2).

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+early research showing that the sampling distribution of the mean quickly approaches normal,
+even when drawing from non-normal distributions—as long as samples are sufficiently large.
+
+It is common to demonstrate this result by sampling from uniform and exponential distributions.
```
Consider joining the two sentences since they're logically linked, and rephrasing

> ...sampling from uniform and exponential distributions: they do not resemble the normal curve...

Alcampopiano replied:
Thank you for the suggestion. I have joined the paragraphs here since they are related. Hopefully [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/bfbc766d3695b76faa51c14cd86ec127c217e291) and an [earlier commit](https://github.com/Alcampopiano/scipy_proceedings/commit/f46499bd0b357b70956619f605c0047f843df7e2) address the concerns sufficiently.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+so-called contaminated normal :cite:`20000755025`:cite:`tan1982sampling`.
+This distribution is also bell-shaped
+and symmetrical but it has slightly heavier tails when compared to the standard normal curve.
+That is, it contains outliers. Consider the distributions in Figure :ref:`contnorm`. The variance of the
```
The standard normal distribution in `contnorm` is a tad difficult to see because it's grey and seems a bit low on alpha: I suggest making a better contrast to the contaminated normal in blue.

The abscissa is also somewhat vague, being labeled "data". If the figure is meant to be talking in the abstract, then perhaps it's better to either omit the ticks and the label, or just drop in `x`.

The normal curve is also truncated at ~| 2.7 | - is there a reason for this?

Alcampopiano replied:
Hello, and thank you for these suggestions. In [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/21e13119c7a9f8e6ead8c8a555c8b98a6e86f6cb), I've improved the colors and removed the X axis label altogether (for both charts). I've also fixed the truncated normal curve. Thanks very much.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+
+The assumption is that T follows a t distribution (and with large samples, approaches normality).
+Figure :ref:`tcurve` tests this assumption by calculating T many times based on a computer-generated
+population distribution (5000 samples; n=20; lognormal distribution).
```
I'm slightly confused by "computer-generated population distribution" - could you please elaborate on the way this is being generated?

Alcampopiano replied:
Thanks very much for this comment. I have greatly improved the readability for this section in [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/f0a244a2bcf6e0663e894fd239d3bafc4d457ab8). That is, I have mentioned much more clearly how the t-distribution is estimated. I hope this reads better from your perspective as well. I am much happier with it now too.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+Student’s t distribution
+------------------------
+
+It turns out that T, the statistic you get from Student’s t-test, is more sensitive to violations
```
This passage in general is phrased a little awkwardly since you start on a new topic without segue, and the subject is out of order.

> Another common statistic is the T-value obtained from the Student's t-test...

Alcampopiano replied:
This was a very helpful suggestion. Not only did I reword [the first sentence](https://github.com/Alcampopiano/scipy_proceedings/commit/4157401e395c6c2b06e0ab633a5e4f7b866587ad) in the section you quoted above but I also added [more detail](https://github.com/Alcampopiano/scipy_proceedings/commit/4409468b60f385adf2d37c125d8d756e471df23f) and references to the preceding section—giving a much better segue. Thank you very much.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+This is despite the fact that the t distribution is also bell shaped, light tailed, and
+symmetrical—a close relative of the normal curve.
+
+The assumption is that T follows a t distribution (and with large samples, approaches normality).
```
t-distribution

Alcampopiano replied:
Thank you. Fixed in [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/1e3a3b6c0f873aca148012cfca1ad9f8ed60260b)

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+
+   Assumed t distribution (grey) and actual t distribution (blue). :label:`tcurve`
+
+One conclusion that can be drawn from this is as follows: When sampling is from skewed
```
> when sampling from

Alcampopiano replied:
Thank you. It does sound nicer that way. Fixed [here](https://github.com/Alcampopiano/scipy_proceedings/commit/e176e336e1493bd1d2ff53d357fa385210dbcc00) in the two occurrences of that phrase.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+   Assumed t distribution (grey) and actual t distribution (blue). :label:`tcurve`
+
+One conclusion that can be drawn from this is as follows: When sampling is from skewed
+distributions (e.g., lognormal), the t-test leads to increased false positive rates (Type I Error).
```
> ...the t-test leads to increase false positive rates (Type I Error), as seen from the left tails in :ref:`tcurve`.

Alcampopiano replied:
Awesome suggestion. I've added it to [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/e176e336e1493bd1d2ff53d357fa385210dbcc00). For beginners, this can be a difficult result to reason about. Your suggestion helps so it is much appreciated.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+distributions (e.g., lognormal), the t-test leads to increased false positive rates (Type I Error).
+You can intuit this by comparing the left tails in the chart above. The middle 95% of the blue
+curve is much wider than that of the gray curve. Therefore, when we assume the t
+distribution in these scenarios, we’re much more likely to incorrectly conclude statistical
```
t-distribution

Alcampopiano replied:
Thank you. Fixed in [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/1e3a3b6c0f873aca148012cfca1ad9f8ed60260b).

laserkelvin commented:
Here is my review for @Alcampopiano's submission:

# Preliminary review

The paper details a common and widespread issue in statistical analysis: the misuse of the normal distribution, often through conflating the idea of it being a mathematical model for some phenomena, and the convergence of all data towards being normally distributed as stated by the central limit theorem. The paper builds on top of this idea with two examples of data that are vaguely similar to being normally distributed, however offer vastly different interpretations if modeled correctly.

The following sections then describe more modern and robust techniques for statistical analysis and inference: notably bootstrap methods, which have gained popularity with the rise of modern computers. The author argues that, despite these concepts being available and at hand, the main obstacle to their wider adoption is a matter of education and tooling: by making these methods more accessible for general researchers and analysts, the broader scientific community would benefit significantly from mitigating the misuse of statistics. This was the motivation behind creating `hypothesize`: a high level Python package designed for notebook environments implements a suite of modern statistical techniques for use by non-experts.

# General comments

The motivation and need for this work is clear: making difficult statistical techniques accessible for broader usage will both help research and training of people who practice statistical analysis, which is a very core skill set in the data-driven age. While the writing is generally good, I find that the paper lacks depth in all sections: the author begins an idea however does not really see it through to connect them properly. The biggest example of this is in the first three sections: the flaws of the normal distribution are discussed, but it's not directly connected with hypothesis testing which is what the paper is _actually_ about; the reader has to make the connection themselves.

Similarly, the description of `hypothesize` comes mostly as some examples of the API, and results that are returned. To add depth to this, I would actually like to see a concrete case study of how `hypothesize` was used, for example as a tutorial for a class, or to solve a research problem. That, I think, forms the decision boundary between an accept and reject: where it stands now, __I do not think the manuscript does the package and its utilization justice.__ Tangentially, I think the examples should also connect with the methods described earlier with _more emphasis_, such as the bootstrap sampling (i.e. connect the high level statistical ideas with concrete usage). I had to re-read the manuscript to see the connection, which is actually kind of abstracted away from the API (a good and a bad thing!).

In short, I think the manuscript has potential, I believe in the package and its motivation, and it will need somewhat major revisions to get it into an acceptable format. At the bare minimum, I would like to see my comment for case studies addressed.

# Questions

1. Are there plans to incorporate Bayesian methods into `hypothesize` as well? In a way, many of the problems with inferring from statistics stems from over-interpretation of frequentist statistics which is very much an education, training, and lack of resources problem as the author described, particularly the "N=40 is enough for normal" scenario. 
2. I noticed that `DataFrames` were referenced in the API examples: how much of the abstraction was designed around `DataFrames`, i.e. does `hypothesize` treat them as a first-class citizen?
3. Is there specific integration of `hypothesize` behind DeepNote (which is proprietary), or are we able to take full advantage of it even in public servers (e.g. Binder) or locally hosted servers?

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+Modern robust methods
+---------------------
+
+When it comes to hypothesis testing, an intuitive way of dealing with the issues described
```
Non-parametric means is one way - another would be to build a better model? I suggest a comment like that should be added.

Alcampopiano replied:
I've changed this to "*one* way to deal with the issues described...". I realize that this doesn't address the addition of mentioning that choosing another model may also be worth while. I am thinking about how to fit that into this section without detracting from the flow here (e.g., setting the ground work to describe the trimmed mean and bootstrapping which comprise the subsequent sections). I will have a think on this though (perhaps a footnote would be perfect).

Alcampopiano replied:
I've also added a footnote with [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/841f34ff60104d76b41b71d7c560c0aacb93599c) that mentions that another underlying model may also be a viable way of dealing with the issues that are presented.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+Non-parametric resampling techniques such as bootstrapping and
+permutation tests build empirical sampling distributions,
+and from these, one can robustly derive p-values and CIs.
+One example is the percentile bootstrap test :cite:`tibshirani1993introduction`.
```
I suggest adding references to the original bootstrap paper by Efron as well

Alcampopiano replied:
Hello @laserkelvin Thank you for this comment. I have added the original Efron paper as suggested.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+5. Consider the middle 95% of all differences (the confidence interval)
+6. If the confidence interval contains zero, there is no statistical difference, otherwise, you can reject the null hypothesis (there is a statistical difference)
+
+The percentile bootstrap is simply a computational algorithm that uses the data at hand to estimate
```
This motivates the use of the bootstrap, and I think would be better to move this before your enumerated list: that way the logic flows from what bootstrapping is, why you would use it, and this is how it's implemented.

Alcampopiano replied:
Great idea. Thanks for the thoughtful suggestion. I have put the high-level description above the enumerated list in [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/0f8fedbd541868a38b8b6c80cf89aade7ff246a9).

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+    from hypothesize.utilities import trim_mean
+    from hypothesize.compare_groups_with_single_factor \
+        import pb2gen
+    results=pb2gen(df.column_1, df.column_2, trim_mean)
```
The code formatting looks a bit awkward here in both the render and in the plain text - I suggest formatting it according to some style guide (e.g. PEP8).

I would also suggest replacing dataframe column references to concrete examples: this relates to my comment about specific case studies with the API: it would be immensely more helpful for students to understand the abstraction if you were to pass columns with names instead of `column_n`.

Alcampopiano replied:
Thanks again for the suggestion. I've made adjustments to the code blocks to conform to PEP8. In addition, I've used column names that will make it the functions clearer and more interesting to the reader. That is, when comparing two groups, I use `df.placebo` and `df.treatment`. I also reference that example in parentheses before showing it in code. Thanks very much for this. Here is [the commit](https://github.com/Alcampopiano/scipy_proceedings/commit/e7fd3632db3682644a984a57beb2c79ddbf751c8).

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+.. code-block:: python
+
+    from hypothesize.measuring_associations import wincor
+    results=wincor(df.column_1, df.column_2, tr=.2)
```
Same comment as in the other code block

Alcampopiano replied:
Resolved (hopefully) in the same manner as the other issue. Here is [the commit](https://github.com/Alcampopiano/scipy_proceedings/commit/e7fd3632db3682644a984a57beb2c79ddbf751c8). Thanks again for the suggestion.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+It is a notebook-based collection of interactive demonstrations aimed at clearly and visually
+explaining the conditions under which classic methods fail relative to robust methods.
+A hosted notebook with the rendered visualizations of the simulations can be accessed
+`here <https://deepnote.com/workspace/allan-campopiano-4ca00e1d-f4d4-44a2-bcfe-b2a17a031bc6/project/robust-stats-simulator-7c7b8650-9f18-4df2-80be-e84ce201a2ff/%2Fnotebook.ipynb)>`_.
```
I actually missed this in the typeset version: the blue highlighting doesn't really differentiate well from the black when it's just a single word, so maybe encompass more text?

laserkelvin replied:
I'd also like to point out that I couldn't get the link to work without actually creating and signing into DeepNote.

Alcampopiano replied:
Thanks for this very helpful catch. I've encompassed more text, and importantly, I had a slightly broken link there which is now fixed [here](https://github.com/Alcampopiano/scipy_proceedings/commit/b1433d685bee342b4c9cfae5bfdcb1752b653ffb). The correct link is to a view-only notebook, and certainly no sign in is required to view the contents. Thanks again for catching that.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,315 @@
+particular estimators and statistical tests have been chosen.
+
+
+.. figure:: sim.png
```
This is actually not an informative figure at all - it's barely mentioned in passing, and I don't know what to make of it other than a nice looking UI. What do you want the reader to take away from this?

Alcampopiano replied:
Thanks for your feedback here. I can remove the figure; however, here is why I believe it is important to show:

I have worked in laboratories and research departments for many years. One thing that I find to be a barrier for those who want to learn about statistics is not having visually appealing and interactive teaching interfaces. When a social science student sees that figure, my sincere hope is that they will say, "That looks easy. Not too much code either. I can do this!". I hope it inspires them to begin learning about this topic. Great learning resources are a very important aspect of what I think is missing from the education of robust statistics and so I'm contributing as best I can.

I've added some [detail](https://github.com/Alcampopiano/scipy_proceedings/commit/da1bfad9988aec136e99fea0a2e6723b63475be1) to "double down" on the fact that these resources are quick and easy ways of onboarding to what is often seen as complex material (despite the fact that much of robust stats is perfectly doable even for those who are a bit rusty with their stats).

Thank you again. I will leave this unresolved and await your thoughts.

laserkelvin replied:
I'm convinced by your point, but I think then you should reference that in your figure caption. Just something to the effect of "it's a simple, minimalist UI so navigation and getting started is easy". I agree that opening up traditional statistical analysis programs like SPSS, Origin, and whatnot the UI is definitely a sensory overload and without proper training is an immediate turn-off for most if not all students.

Alcampopiano replied:
Thanks for your understanding here. I have [updated the figure caption](https://github.com/Alcampopiano/scipy_proceedings/commit/b64cd7be1285c6b289e1d7be057f2439b2c57d9e) as mentioned.

deniederhut commented:
@scoobies mark needs more review

deniederhut commented:
Hey @ojemebo! Friendly reminder that the deadline for initial reviews is tomorrow

deniederhut commented:
@scoobies remove @ojemebo from reviewers

scoobies commented:
@ojemebo removed from the reviewers list!

deniederhut commented:
@scoobies mark pending comment

Alcampopiano committed:
addressing comment https://github.com/scipy-conference/scipy_proceedings/pull/717#discussion_r897141230. Thank you. I have added a reference to the original paper by Efron

Alcampopiano committed:
addressing comment https://github.com/scipy-conference/scipy_proceedings/pull/717#discussion_r897142041. Adding bootstrap motivation before enumerated list for readability

Alcampopiano committed:
addressing comment https://github.com/scipy-conference/scipy_proceedings/pull/717#discussion_r893628969. I have reworded this to be more formal (and improve clarity in general

Alcampopiano committed:
addressing comment https://github.com/scipy-conference/scipy_proceedings/pull/717#discussion_r893631413. Joined related paragraphs.

Alcampopiano committed:
addressing comment https://github.com/scipy-conference/scipy_proceedings/pull/717#discussion_r893657783. Improved colors, fixed truncated line for notmal dist, and removed X axis labels

Alcampopiano committed:
addressing comment https://github.com/scipy-conference/scipy_proceedings/pull/717#discussion_r893660495. Thank you. I have made the process of estimating the t-distribution explicit. The hop is that others can now more clearly replicate this method

Alcampopiano committed:
adding hyphen to t-distribution

Alcampopiano committed:
grammar and improved clarity when discussing Type I Error result in the t-distribution figure

Alcampopiano committed:
fixed the broken link in hosted notebook example, and also set to read only to make sure no sign in is required

Alcampopiano committed:
improving flow and adding more detail before the section on Student's T; added refs for Hill, and Wilcox 90a.

Alcampopiano committed:
based on suggestion from @laskerkelvin, I've improved the first sentence for clarity

Alcampopiano committed:
adding more humanreadable column names when giving examples of Hypothesize. PEP8 formating code blocks

Alcampopiano committed:
added more detail around the figure of the robust statistics simulation notebook

Alcampopiano committed:
very slight word change to lighten the tone up here, making room for other options to deal with non-robustness

Alcampopiano committed:
added a footnote that mentions that a parametric test with another model assumtion may be another viable way to deal with the issues presented

deniederhut commented:
Thanks for the revisions @Alcampopiano ! Is this paper ready for Kelvin to take a second look?

Alcampopiano commented:
> Thanks for the revisions @Alcampopiano ! Is this paper ready for Kelvin to take a second look?

Hello @deniederhut, I must still address Kelvin's comments and questions at the top of this thread. I am planning on doing that early this week. Take care and thanks for the ping.

Alcampopiano committed:
Three major revisions: 1. Student's t section (adding actual Type I error stats and improved discussion); 2. Contaminated normal section (adding power simulation research and new figure); 3. Added new case study section with real-world data comparing robust and traditional methods (with code examples)

Alcampopiano committed:
fixed power figure and typos in caption

Alcampopiano committed:
added citations of Hypothesize being used in applied research

Alcampopiano committed:
added explicit mention of Pandas DF, linked to docs, and clarified that Hypothsize can be used in any computer environment

Alcampopiano commented:
Hello @laserkelvin and @deniederhut,

Thank you both for you work on this submission. I have addressed each of the individually suggested changes above to the best of my ability. In addition, I would like to now address each of the comments and questions raised by @laserkelvin in the [preliminary review section above](https://github.com/scipy-conference/scipy_proceedings/pull/717#issuecomment-1155524746):

## Addressing the general comments
@laserkelvin Thank you for your thoughtful comments in the “general comments” section above. Please let me enumerate them here and explain how I’ve addressed them in the paper:

**1. Lack of depth in introductory sections**

In [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/41db6346555285cbb101906da046eaaefbfee7df) (and others) I have added substantial depth to the section on contaminated normals and the section on Student’s t-test. For example:
- I have clarified why assuming the t-distribution may lead to increased Type I error. Rather than simply saying “you can intuit this from the tails of the empirically derived t-distribution” (figure 3), I’ve listed out the approximate critical values (beyond which, one would reject the null) and highlighted the wider-than-expected 95% CIs. I think this reads much better and connects the dots for the reader more explicitly.
- Instead of only mentioning the effect on variance when sampling from a contaminated distribution, I have added a new figure and discussion around how power and hypothesis testing are affected by contamination when compared to normally distributed groups. It is the striking reduction of power, even when distributions appear to be normal, that I’ve done my best to underscore in this new addition.

**2. Poor connection between high level ideas in the introduction and the solutions presented in code.**

- To connect later sections with the improvements above, I’ve added a case study using Hypothesize with real-world data on sexual attitudes (the data were collected by Miller and are available on Rand R. Wilcox's site). With this dataset, I provide a comparison of traditional methods to robust methods when comparing independent groups (this aligns with previous discussions in the paper on power and hypothesis testing). In the example I provide, we can reject the null with Hypothesize but not with a traditional t-test. I also provide citations pointing to (1) comparisons of this sort in the literature and (2) where Hypothesize is being used in peer-reviewed applied research. Other details are given that discuss hypothesis testing and why researchers should not necessarily expect to reject the null with robust methods—further connecting them to earlier points in the paper on Type I Error.

**3. Lack of a concrete case study or real-world example of Hypothesize**
- As mentioned above, the code and data are available to reproduce the case study that I’ve provided for Hypothesize. In addition, citations where Hypothesize has been used in applied research have been given. This, in conjunction with the basic usage examples, should do a better job of connecting the introductory issues I raise to my software—underscoring how robust methods can make a practical difference with real-world data.

## Addressing questions:

> Are there plans to incorporate Bayesian methods into hypothesize as well? In a way, many of the problems with inferring from statistics stems from over-interpretation of frequentist statistics which is very much an education, training, and lack of resources problem as the author described, particularly the "N=40 is enough for normal" scenario.

There are no plans in the short term to implement Bayesian methods into Hypothesize. For now, my goal is to reach parity with Wilcox’ WRS package. Consistency between the two packages is likely a good thing for researchers moving from R to Python (and back).

> I noticed that DataFrames were referenced in the API examples: how much of the abstraction was designed around DataFrames, i.e. does hypothesize treat them as a first-class citizen?

These are actually Pandas DataFrames and so there is no additional abstraction specific to Hypothesize. I have now mentioned this explicitly in the paper (the Hypothesize docs also make this clear). Thanks for pointing this out.

> Is there specific integration of hypothesize behind DeepNote (which is proprietary), or are we able to take full advantage of it even in public servers (e.g. Binder) or locally hosted servers?

Hypothesize and the robust status simulator are open-source Python packages. They can be used in any IDE or notebook environment. It may be easier for some to have a “click and go” option (say, for teaching), but certainly there is nothing preventing that code from being used outside of Deepnote. I have mentioned this explicitly now in the paper and re-highlighted that Hypothesize is open source (please see [this](https://github.com/Alcampopiano/scipy_proceedings/commit/689a3eed6f7f31b0c00a217eecf5dc53e5809b37) commit).

---

I would again like to thank @laserkelvin for both the positive and constructive feedback on the initial version of this paper. I am very much pleased with the improvements and hope that they satisfy any major issues that were present. 

Sincerely,
Allan

Alcampopiano committed:
move yuenbt parameters to new line to prevent code getting cut off by the typesetting

Alcampopiano committed:
remove axis titles and main titles from figure 2

Alcampopiano committed:
fixed typos and added more active citation styles where needed

laserkelvin commented:
Thanks a lot @Alcampopiano for putting in all the effort of making the changes!

I'll take a look at it as soon as I can, but unfortunately got COVID from a conference last week so I'll be a bit slower...

cbcunc commented:
Reviewers, please give your final recommendations. The deadline for final review comments is tomorrow.

Authors, please respond to review comments with changes or comments which resolve the conversation. Final acceptance decisions are July 4.

laserkelvin commented:
@cbcunc I'm not working at 100% due to COVID, so that deadline is unfeasible for me. I can try my best to give my final recommendation a day later (7/1).

deniederhut commented:
😬  best wishes for your recovery Kelvin.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,418 @@
+are not discussed in this article. Interested readers should see Wilcox's
+1992 paper :cite:`wilcox1992can`.
+
+.. figure:: power_concat.png
```
In perfect honesty, I don't think this figure does your message justice: if the point is to show a minute, basically invisible difference then having them side-by-side is not the best way to visualize them. I suggest using shading to indicate how the statistical power changes, or at least some way to more strongly pronounce that in the figure.

laserkelvin replied:
Also, the fact that the two plots are not on the same y-axis makes it even more difficult to tell the difference.

Alcampopiano replied:
Thanks for bringing this up as I have often felt the same. Ideally, I am trying to convey all of the following: (1) power when comparing groups (say, with Student's t) and (2) how very similar looking pairs of curves can produce wildly different results. 

It is not easy getting this across in a figure. But I have done some simple improvements that I think are sufficient.

Firstly, I have added a [much better figure caption](https://github.com/Alcampopiano/scipy_proceedings/commit/650de8875ef7101138d53e9020ac974b179415a9). This alone makes both points I mentioned above to support the image.

Secondly, as you rightly mentioned, I [set the Y limits](https://github.com/Alcampopiano/scipy_proceedings/commit/41ea34bb594c76c7c4e74cb08e5f246d30c27b00) so that they are equal. This is a much fairer representation of the data since it highlights the differences between the two sets of curves (even though they are similar in many ways—which is the point I'm going for).

I hope this are enough to satisfy the suggestion. I am quite happy with this improvement.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,418 @@
+and (2) statistics other than the sample mean were not tested and may behave differently
+(e.g., Student’s t). In the half century following these early findings, many important
+discoveries have been made :cite:`wilcox2013introduction`. These call into question the usefulness of
+the normal curve and will be discussed in the following sections.
```
In lieu of an abstract, I would suggest alluding to Hypothesize at the beginning: the point of this paper is to draw attention to standard problems in statistical analyses, and you want the reader to know there is a solution you've developed - otherwise it's difficult to know where the paper is going as it stands.

Alcampopiano replied:
Great improvement. I've added a [nice little section](https://github.com/Alcampopiano/scipy_proceedings/commit/85d6947db08cd9a2472e52a8ad4a78a0a4831dd8) setting the stage accordingly now.

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,418 @@
+   :scale: 50%
+   :align: center
+
+   Standard normal (orange) and contaminated normal (blue). :label:`contnorm`
```
As a matter of personal choice, I subscribe to the idea of self-contained figures and figure captions: I should be able to understand what the figure shows me and its immediate context by looking at the plot and its caption. In this case, I'd suggest adding something short to relate to what you say in the text, like "A small difference in the peak probability density changes the variance significantly"

Alcampopiano replied:
I think I may begin taking your perspective on figure captions in general. Now that I've expanded them to be self contained in terms of explanation. I like them so much better :).

Thanks very much. You can see the commit [here](https://github.com/Alcampopiano/scipy_proceedings/commit/650de8875ef7101138d53e9020ac974b179415a9) (all figures captions have been updated accordingly).

laserkelvin replied:
I think you did a great job with them - the captions really add context to the figures _and_ the text now and help tie everything together 😄

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,418 @@
+use Hypothesize can be relatively straightforward. In fact, every statistical
+test listed `in the docs <https://alcampopiano.github.io/hypothesize/>`_
+is associated with a hosted notebook, pre-filled with
+sample data and code. But certainly, simply `pip install Hypothesize` to
```
`pip install hypothesize` isn't rendering correctly

Alcampopiano replied:
Thanks! I had made a syntax mistake. It is fixed in [this commit](https://github.com/Alcampopiano/scipy_proceedings/commit/0c7f5df3cb46c50e168ac1c037d1f8d04bacdb79).

laserkelvin commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,418 @@
+van Noordt et al. :cite:`van2022inter`, and Elsabbagh :cite:`elsabbagh2022inter`.
+
+The API for Hypothesize is organized by single- and two-factor tests, as well as
+measures of association. Data is always in the form of Pandas DataFrame.
```
Reference for Pandas?

Alcampopiano replied:
Thank you. Apparently there's two (software + paper) and now [they are both present](https://github.com/Alcampopiano/scipy_proceedings/commit/cf904d146718d94de9f6b92a89e68082587bef9c) in my paper.

laserkelvin reviewed (approved):
Thank you @Alcampopiano for making all the effort to do all the changes I've asked: with the expanded section on real life applications, as well as elaborating further on a few points especially in the early parts of the paper, I believe this manuscript is acceptable for publication!

I've left a small number of comments throughout that I think will improve the paper further - most generally, I think the figure captions should be expanded a bit to include more contextual information rather than just a statement of what it is to help readers connect the dots further.

Great work, and it's been a pleasure to read and review this manuscript of yours! Unfortunately my line of work doesn't need Hypothesize immediately, but I'll be sure to point students in that direction if they need to do some robust statistics!

TL;DR Final recommendation is :shipit:

cbcunc commented:
@Alcampopiano Would you please respond with changes to these last few final review comments by July 4?

cbcunc commented:
> @cbcunc I'm not working at 100% due to COVID, so that deadline is unfeasible for me. I can try my best to give my final recommendation a day later (7/1).

Thanks for your help @laserkelvin and I hope you come back to 100% soon,

Alcampopiano committed:
added a mention of my software in lieu of an abstract

Alcampopiano committed:
fix inline code display issue

Alcampopiano committed:
added Pandas citations and improved wording

Alcampopiano committed:
beter figure caption for the UI figure

Alcampopiano committed:
improved figure captions

Alcampopiano committed:
even more improved figure captions

Alcampopiano committed:
fixed delta capital

Alcampopiano committed:
equal Y axis for power figure

Alcampopiano commented:
Hello @cbcunc, @deniederhut, and @laserkelvin,

I have completed the remaining (and very helpful) suggestions. Thank you for the time you have given and please let me know if there is anything else you need from me.

deniederhut commented:
Thanks @Alcampopiano ! @cbcunc is this paper ready?

deniederhut commented:
@cbcunc is this paper ready for publication?

Alcampopiano committed:
improvements to grammar

cbcunc commented:
@scoobies mark ready

deniederhut commented on papers/allan_campopiano/paper.rst:
```diff
@@ -0,0 +1,434 @@
+The author would like to thank Karlynn Chan and Rand R. Wilcox as well
+as Elizabeth Dlha and the entire Deepnote team for their support
+of this project. In addition, the author would like to thank
+Kelvin Lee for his insightful review of this manuscript.
```
❤️

Alcampopiano replied:
❤️ Indeed.

deniederhut commented on papers/allan_campopiano/mybib.bib:
```diff
@@ -0,0 +1,239 @@
+    publisher={Chapman and Hall New York}
+}
+
+@article{erceg2008modern,
```
Is this missing a DOI?

Alcampopiano replied:
Done! As well as all the others too. Thank you.

deniederhut commented on papers/allan_campopiano/mybib.bib:
```diff
@@ -0,0 +1,239 @@
+  publisher={Wiley Online Library}
+}
+
+@article{elsabbagh2022inter,
```
I think we're missing a journal name and volume for this one?

Alcampopiano replied:
Thanks very much. That was actually a duplicate of another citation so it has been removed!

deniederhut commented on papers/allan_campopiano/mybib.bib:
```diff
@@ -0,0 +1,239 @@
+  publisher={Elsevier}
+}
+
+@article{van2022inter,
```
Also missing a DOI?

Alcampopiano replied:
Done! As well as all the others too. Thank you!

deniederhut reviewed (changes requested):
Please take a second pass over your references section -- some of them look incomplete

deniederhut commented:
@scoobies mark pending comment

Alcampopiano committed:
fixed DOIs and added URLs where necessary

Alcampopiano commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1037/0033-2909.105.1.156 is OK
- 10.1080/01621459.1993.10594325 is OK
- 10.1016/c2010-0-67044-1 is OK
- 10.1111/1467-8721.ep10768801 is OK
- 10.1037/0003-066X.53.3.300 is OK
- 10.1177/016059768100500103 is OK
- 10.1016/j.bbr.2018.03.025 is OK
- 10.1037/0003-066X.63.7.591 is OK
- 10.1007/978-1-4612-4380-9_41 is OK
- 10.2307/2530452 is OK
- 10.1348/000711008X325300 is OK
- 10.2307/2334299 is OK
- 10.1016/j.dcn.2021.100945 is OK
- 10.1002/aur.2701 is OK
- 10.5281/zenodo.3509134 is OK
- 10.25080/Majora-92bf1922-00a is OK

MISSING DOIs

- None

INVALID DOIs

- 0.21105/joss.02241 is INVALID
```

Alcampopiano committed:
fixed Hypothesize DOI

Alcampopiano commented:
@scoobies check references

scoobies commented:
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1037/0033-2909.105.1.156 is OK
- 10.1080/01621459.1993.10594325 is OK
- 10.1016/c2010-0-67044-1 is OK
- 10.1111/1467-8721.ep10768801 is OK
- 10.1037/0003-066X.53.3.300 is OK
- 10.1177/016059768100500103 is OK
- 10.1016/j.bbr.2018.03.025 is OK
- 10.1037/0003-066X.63.7.591 is OK
- 10.21105/joss.02241 is OK
- 10.1007/978-1-4612-4380-9_41 is OK
- 10.2307/2530452 is OK
- 10.1348/000711008X325300 is OK
- 10.2307/2334299 is OK
- 10.1016/j.dcn.2021.100945 is OK
- 10.1002/aur.2701 is OK
- 10.5281/zenodo.3509134 is OK
- 10.25080/Majora-92bf1922-00a is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

Alcampopiano commented:
Hello @deniederhut 👋,

Thank you very much for pointing out the issues in the reference section. I've combed through every reference and added DOIs (or URLs) where necessary. 

I hope things are in tip-top shape now. Please let me know if you need anything else.

deniederhut reviewed (approved):
Thanks!

deniederhut commented:
@scoobies mark ready
